packages/react/src/components/ChatHeader/ChatHeader.js

```diff
@@ -155,29 +155,44 @@ const ChatHeader = ({
     setShowSearch(false);
   }, [setShowMentions, setShowSearch]);
 
+  const authenticatedUserRoles = useUserStore((state) => state.roles);
+  const setCanPinMsg = useUserStore((state) => state.setCanPinMsg);
+
   useEffect(() => {
     const getMessageLimit = async () => {
       const messageLimitObj = await RCInstance.getMessageLimit();
       setMessageLimit(messageLimitObj?.value);
     };
 
-    const setMessageAllowed = async () => {
+    const setMsgAndPinAllowed = async (ro) => {
       const permissionRes = await RCInstance.permissionInfo();
       const channelRolesRes = await RCInstance.getChannelRoles(
         isChannelPrivate
       );
 
       if (permissionRes.success && channelRolesRes.success) {
+        const pinMsgRoles = permissionRes.update[146]?.roles || [];
         const postMsgRoles = permissionRes.update[140]?.roles || [];
 
-        const userRoles = channelRolesRes.roles
+        const channelMemberRoles = channelRolesRes.roles
           .filter((chRole) => chRole.u?._id === authenticatedUserId)
           .flatMap((chRole) => chRole.roles);
 
-        const canSendMsg =
-          userRoles.length > 0 &&
-          postMsgRoles.some((role) => userRoles.includes(role));
-        setCanSendMsg(canSendMsg);
+        const concatenatedRoles = channelMemberRoles.concat(
```

Comment:
You can use shorthand here, destructure and combine. I have raised a PR feat: Addition of chatinfo to display status, have a look here..

Reply:
Get @Spiral-Memory, if we destructure and combine, the users are unable to login